starter-code/src/App.js

```diff
@@ -1,21 +1,80 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+import "./App.css";
+import contacts from "./contacts.json";
+import Info from "./all-info";
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { allContacts: contacts.slice(0, 5) };
+  }
+
+  randomContact() {
+    /* Dnt frgt!!:The push() method adds one or more elements to the end of an array and returns the new length of the array. */
+    this.state.allContacts.push(
+      contacts[Math.floor(Math.random() * contacts.length)]
+    );
+    // floor tam sayı içindi!
+    // pour recuperer allcontacts:
+    this.setState({ listContacts: this.state.allContacts });
+  }
+
+  sortPopularity() {
+    this.state.allContacts.sort(function(celebrity1, celebrity2) {
+      if (celebrity1.popularity < celebrity2.popularity) return -1;
+      if (celebrity1.popularity > celebrity2.popularity) return 1;
+      else return 0;
+    });
+    this.setState({ listContacts: this.state.allContacts });
+  }
+
+  sortName() {
+    this.state.allContacts.sort(function(celeb1, celeb2) {
+      if (celeb1.name < celeb2.name) return -1;
+      if (celeb1.name > celeb2.name) return 1;
+      else return 0;
+    });
+    this.setState({ listContacts: this.state.allContacts });
+  }
+
+  delete = name => {
```

Comment:
your delete doesn't work because you've called the function here delete, but the function that is linked to your button is `deleteInfo` and your state should be `    this.state.allContacts.splice(name, 1);`